traits/trait_types.py

```diff
@@ -1516,6 +1526,9 @@ def create_editor(self):
 class Directory(BaseDirectory):
     """ A fast-validating trait type whose value is a directory path string.
 
+    For Python 3.6 and greater, it also accepts objects implementing
+    the `os.PathLike` interface; converting them to corresponding string.
```

Comment:
Grammar nit: replace the semicolon with a comma here (and perhaps remove the first comma in the sentence). Add "the" before "corresponding string".